src/CamelKNodeProvider.ts

```diff
@@ -241,22 +244,20 @@ export class CamelKNodeProvider implements vscode.TreeDataProvider<TreeNode> {
 			console.log('Command string: ' + commandString);
 			let runKamel = child_process.exec(commandString);
 			var shellOutput = '';
-			if (runKamel.stdout) {
+			if (runKamel.stdout && runKamel.stderr) {
 				runKamel.stdout.on('data', function (data) {
 					console.log("[OUT] " + data);
 					shellOutput += data;
 				});
-			}
-			if (runKamel.stderr) {
 				runKamel.stderr.on('data', function (data) {
 					console.log("[ERROR] " + data);
 					reject(data.toString());
 				});
+				runKamel.on("close", () => {
```

Comment:
can you explain the advantage of including the 3 actions in the same if branch?
Looks alike we are loosing the case one of the two stout or stderr still available that wants to log something. And also, is the "close" still logged? stdout and stderr are still available when the event is triggered?

Reply:
I honestly can't recall why I shifted it to a single branch at this point. I'll shift it back and see what happens.

Reply:
Looking at this again, it has nothing to do with this PR - so I'll spin this and the other comment off into a separate issue to investigate.